### PR TITLE
Fork choice fixes 2

### PR DESCRIPTION
**Pull request by arnetheduck**

### CODE REVIEW

File: beacon_chain/spec/helpers.nim [hunk 1]
```diff
@@ -56,6 +56,12 @@ func is_active_validator*(validator: Validator, epoch: Epoch): bool =
   ### Check if ``validator`` is active
   validator.activation_epoch <= epoch and epoch < validator.exit_epoch
 
+# https://github.com/ethereum/eth2.0-specs/blob/v0.12.1/specs/phase0/beacon-chain.md#get_active_validator_indices
```
Comment (tersec) [inline beacon_chain/spec/helpers.nim:59]: It's a fine abstraction, but still not something one would want to call that often, for the same reasons that `get_active_validator_indices()` is not great to call often.

Comment (tersec) [reply beacon_chain/spec/helpers.nim:59]: Also, as an isolated function, just `count_active_validators()` captures the idea fine, as a name. There is, though, a case for the symmetry between `{get,count}_active_validator_indices()`. Subjective and to taste.

File: beacon_chain/block_pools/candidate_chains.nim [hunk 2]
```diff
@@ -60,9 +60,10 @@ func parent*(bs: BlockSlot): BlockSlot =
       slot: bs.slot - 1
     )
 
-func populateEpochCache(state: BeaconState, epoch: Epoch): EpochRef =
-  (EpochRef)(
-    epoch: state.slot.compute_epoch_at_slot,
+func populateEpochCache(state: BeaconState): EpochRef =
```
Comment (tersec) [inline beacon_chain/block_pools/candidate_chains.nim:63]: Yeah, I noticed this when investigating possible causes for the recently found cache invalidation bug. it's conceptually plausible to have that flexibility of an epoch distinct from the state epoch, but in practice was unused.

File: beacon_chain/spec/helpers.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/spec/helpers.nim:59]: hm, yeah, I suspect there may be tricks to get rid of it - until then, it's a trivial addition that simplifies some code

Comment (tersec) [reply beacon_chain/spec/helpers.nim:59]: It also avoids memory allocations, which is probably much of what's left outside BLS cryptography.

**Pull request merged**